### ISSUE TITLE
P2020有问题，射击会使延迟增高游戏崩溃

### ISSUE THREAD

**Issue by GITDK2022** [mgsweet/Apex-NoRecoil-2021#305]
P2020有问题，射击会使延迟增高游戏崩溃

**Comment by 010315wc** [comment]
yes

**Comment by mggtx** [comment]
me too

**Comment by reibc** [comment]
#307 should fix it

**Comment by TedzNoodle** [comment]
can i ask how to use autofile ? use hotkey to turn on or what, and can hemlock autofire ? 
i try any key button to turn on but cant? 
i don't know much of the code ?

**Comment by RaphielHS** [comment]
> > can i ask how to use autofile ? use hotkey to turn on or what, and can hemlock autofire ? i try any key button to turn on but cant? i don't know much of the code ?
> 
> auto-fire seems to work for p2020 & hemlock(unless it's bugged), when u hold left click, it will automatically spam click for u. If it doesn't work, try to press 1 or 2 (depending on which weapon slot the weapon is in) and then try holding left click again

Yeah, Hemlok is bugged, i added some weapon myself by editing the code that decides if it should enable autofire or no at line 463 on the function "IsAutoClickNeeded"

I added for myself: Prowler (works well), Hemlok (Buggy)

**Comment by TedzNoodle** [comment]
> > can i ask how to use autofile ? use hotkey to turn on or what, and can hemlock autofire ? i try any key button to turn on but cant? i don't know much of the code ?
> 
> auto-fire seems to work for p2020 & hemlock(unless it's bugged), when u hold left click, it will automatically spam click for u. If it doesn't work, try to press 1 or 2 (depending on which weapon slot the weapon is in) and then try holding left click again

i try, i change the code like #307, but nothing happen, it seem bugged agian ?, chaging weapon, test every button, but nothing.

in last patch befor #307, it seem 1 bug like if i hold left click, and use B (changing fire mode), it auto even i dont left click, but it seem terrable.

**Comment by TedzNoodle** [comment]
> > > can i ask how to use autofile ? use hotkey to turn on or what, and can hemlock autofire ? i try any key button to turn on but cant? i don't know much of the code ?
> > 
> > 
> > auto-fire seems to work for p2020 & hemlock(unless it's bugged), when u hold left click, it will automatically spam click for u. If it doesn't work, try to press 1 or 2 (depending on which weapon slot the weapon is in) and then try holding left click again
> 
> Yeah, Hemlok is bugged, i added some weapon myself by editing the code that decides if it should enable autofire or no at line 463 on the function "IsAutoClickNeeded"
> 
> I added for myself: Prowler (works well), Hemlok (Buggy)

and can you use trigger, i dont know how to use too?, can u show me ?

**Comment by TedzNoodle** [comment]
> > can i ask how to use autofile ? use hotkey to turn on or what, and can hemlock autofire ? i try any key button to turn on but cant? i don't know much of the code ?
> 
> auto-fire seems to work for p2020 & hemlock(unless it's bugged), when u hold left click, it will automatically spam click for u. If it doesn't work, try to press 1 or 2 (depending on which weapon slot the weapon is in) and then try holding left click again

yeah, and p2020 can, but hemlok cant

**Comment by reibc** [comment]
> > > can i ask how to use autofile ? use hotkey to turn on or what, and can hemlock autofire ? i try any key button to turn on but cant? i don't know much of the code ?
> > 
> > 
> > auto-fire seems to work for p2020 & hemlock(unless it's bugged), when u hold left click, it will automatically spam click for u. If it doesn't work, try to press 1 or 2 (depending on which weapon slot the weapon is in) and then try holding left click again
> 
> yeah, and p2020 can, but hemlok cant

In code, the auto-fire option is implemented only for hemlok single-fire mode
